### ISSUE TITLE
[Package]: SDL3

### ISSUE THREAD

**Issue by DuilioPerez** [termux/termux-packages#19738]
### Why is it worth to add this package?

This new version of SDL adds some new functionalities, like filesystem handling, file request through dialog boxes and time and date utilities. It has been prereleased.

### Source code URL

https://github.com/libsdl-org/SDL

### Packaging policy acknowledgement

- [x] I certify that I have read [Termux Packaging Policy](https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#packaging-policy) and understand that my request will be denied in case of violation.

**Comment by Biswa96** [comment]
SDL3 is still in preview release.

**Comment by DuilioPerez** [comment]
Yes, I know it. But there are builds available on its official github repo for linux and windows. It could be great if is added to Termux.

**Comment by DuilioPerez** [comment]
SDL3 has been officially [released](https://github.com/libsdl-org/SDL/releases/tag/release-3.2.0) now!

**Comment by DuilioPerez** [comment]
I compiled it and works fine!

https://github.com/user-attachments/assets/b146cbf9-0305-4881-b6b5-446d29e941a9

**Comment by truboxl** [comment]
I believe all existing SDL 1 and SDL 2 packages can use sdl2-compat to use SDL 3 for testing purposes.

Some SDL 2 components not yet have SDL 3 counterparts so I am leaving this issue opened for now.

**Comment by CorruptVoidSoul** [comment]
Coming here to say I'm interested and also to give a bit of a status update.
So we have SDL3 in the Termux repos, that's one down.
SDL3_image is now stable, so can be packaged.
SDL3_ttf is in ABI preview state, it'll probably be next.
SDL3_mixer and SDL3_net, well, they don't seem to be in ABI preview for now.

And of course thanks a bunch truboxl for being willing to do that 👍

**Comment by DuilioPerez** [comment]
We have ported SDL_gfx too for SDL3. I've made some contribution and a release in my fork. I've compiled it in Termux and it works perfect. I've made a release in my port and asked today I've asked permission to the official mantainer to do it official. I'll write here when it'll get officially release. But if you are interested, this is the link to [my fork]()

**Comment by Biswa96** [comment]
Please keep this point in mind. "standalone library packages are of interest mostly for developers and we will not package them unless required as dependency for another package." - https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#packaging-policy

**Comment by DuilioPerez** [comment]
> Please keep this point in mind. "standalone library packages are of interest mostly for developers and we will not package them unless required as dependency for another package." - https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#packaging-policy

Yes, I know. I just write it because there is the package sdl2-gfx in Termux. Basically sdl_image, sdl_ttf, sdl_mixer, sdl_net and sdl_gfx are always listed as extensions of sdl, even in emscripten

**Comment by DuilioPerez** [comment]
Excuse me by the question, but can you add a properly configured cmake file to find it. I installed from the Termux repositories, and I can't find it with CMake. I compiled it at hand with CMake support, and I believe it could be usefull. When I try to find it with CMake, it halts with an error. Of course it find a cmake find file, but that file is from my first build, so it doesn't find the corresponding library files. I know this happens with a lot of libraries in the Termux repositories, they can't be find from CMake. I'll understand if is not possible.

**Comment by Biswa96** [comment]
> I installed from the Termux repositories, and I can't find it with CMake.

That should be in a new issue.

**Comment by TomJo2000** [comment]
This should have been closed with the merger of #23038.

**Comment by stsp** [comment]
Still no sdl3_ttf?
Any plans?

**Comment by Biswa96** [comment]
> Still no sdl3_ttf? Any plans?

"standalone library packages are of interest mostly for developers and we will not package them unless required as dependency for another package." - CONTRIBUTING.md

**Comment by DuilioPerez** [comment]
> > Still no sdl3_ttf? Any plans?
> 
> "standalone library packages are of interest mostly for developers and we will not package them unless required as dependency for another package." - CONTRIBUTING.md

It seems SDL3 libraries are being now used in some packages. If I'm not wrong, @stsp needs it for dosemu2

**Comment by stsp** [comment]
As a hard requirement it will
not be the case for the near
future. But you can switch
many packages from sdl2 to
sdl3 if you want, because they
already support both.
If you don't start switching -
there is never a dep, and never
a "need" to include sdl3...

**Comment by truboxl** [comment]
Reopening because not all SDL3 packages are added. Dont understand why it was closed in the first place. https://github.com/termux/termux-packages/issues/19738#issuecomment-2617946683

**Comment by DuilioPerez** [comment]
> Reopening because not all SDL3 packages are added. Dont understand why it was closed in the first place. [#19738 (comment)](https://github.com/termux/termux-packages/issues/19738#issuecomment-2617946683)

It was closed by this:


> > Still no sdl3_ttf? Any plans?
> 
> "standalone library packages are of interest mostly for developers and we will not package them unless required as dependency for another package." - CONTRIBUTING.md

All SDL3 related libraries are seen like "standalone library packages"
